python/doc/content/python/MooseDocs/extensions/tag.md

```diff
@@ -0,0 +1,54 @@
+# Tagging Extension
+
+The tagging extension provides the "tag" command, which provides a mechanism for setting a set of
+tags made up of arbitrary key:value pairs for categories/labels associated with a given markdown page.
+The provided metadata is then injected into a compatible user-provided JavaScript file, which then
+allows for the filtering of tagged pages against the defined categories in the generated website.
+The configuration of the extension contains an optional allowed key list, which is checked against
+during the invocation of the command. The available configuration items for this extension are provided
+in [tagging-config].
+
+!alert! warning title=The tagging extension is experimental!
+This extension is considered to be in an experimental state, as several manual steps are required in
+order to obtain both a set of tagged pages, and a rendered page for filtering against those tags. If
+documentation tagging features are desired, please request assistance from the MOOSE Framework
+development team.
+!alert-end!
+
```

Comment:
is there a page online for MOOSE that we can link to? that shows this active for a few things?

Reply:
There is not with this PR - I could add one to the electromagnetics module (move a couple commits from my other branch to this one) if you want a sample that will be online regularly. Adding it to the test dirs for MooseDocs wouldn't end up being published online.

Reply:
Once the doco rebuilds, an example page should be live at https://mooseframework.inl.gov/docs/PRs/25388/site/modules/electromagnetics/filter/index.html

Reply:
seems it s not working

Reply:
Looks like an issue with the HTML. Browser is looking in `https://mooseframework.inl.gov/js/index-93b559a6.js`, which obviously doesn't exist. [The actual location](https://mooseframework.inl.gov/docs/PRs/25388/site/js/index-93b559a6.js) does however and contains the right information. Will need to explore this tomorrow.

Reply:
Good to go now at the link I posted earlier. Moved to manual relative paths within the HTML, consistent with others that MooseDocs generates.

Reply:
so the filters arent quite combining the way we wanted.

for example, if you specify benchmark and 2021, we would expect to see less results (only the benchmarks from 2021), not the benchmarks + the models contributed in 2021 (so more results)

Reply:
That would be a @friedmud question. The filtering functionality itself is within his tagview repository and outside of this PR.

Reply:
Also I agree with what you are saying - I would also like to see further filtering with more categories selected.